packages/tools/examples/videoSegmentation/index.ts

```diff
@@ -224,11 +224,14 @@ async function run() {
   // We need the map on all image ids
   const allImageIds = viewport.getImageIds();
   const firstImage = allImageIds[0];
-  const segImages = await imageLoader.createAndCacheDerivedImages(
-    [firstImage],
+  const segImages = await imageLoader.createAndCacheDerivedLabelmapImages(
+    allImageIds,
     {
       skipCreateBuffer: true,
       onCacheAdd: csUtils.VoxelManager.addInstanceToImage,
+      getDerivedImageId: (referenceImageId, index) => {
```

Comment:
Please use the multiframe utilities to update the frame reference.
Also, there is no need to use a random uid, you can use the original uid with derived in front of it, but updating the frame index so that every time you generate the same uids.
Finally, this derivation should probably go into the core code so that the handling is consistent and can be done from things like OHIF.

Reply:
@sedghi  - should we add a vewport call 'getDerivedImageIds'   It could take a range request to allow getting a sub-range of it.

Reply:
https://github.com/cornerstonejs/cornerstone3D/blob/0b67f91a41eb069e0ef212d4be7acdcb4c68c542/packages/core/src/loaders/imageLoader.ts#L356-L368

line 360 is creating a derivedImageId using a random uuid anyway if we don't implement `getDerivedImageId`. so i am little confused.

here's a trace:
createAndCacheDerivedLabelmapImages
createAndCacheDerivedImages
line 360

hene i assumed i was not doing anything drastically different reg the creation of a derivedImageId. i thought `getDerivedImageId` lets the user create a derviedImageId in a format they want, which is why i used it to append frames to the id.

Reply:
@wayfarer3130

Reply:
Could you at least move the create derived images into a specialized method in imageLoaders.ts that takes a type argument that can be video/stack/volume/wsi and returns the right type of response?  It should take the same type of argument as the rendering engine create does.  That way the same code can be re-used more generally, and we can update it to use non-random uids if we choose, or continue as is.

Reply:
that makes more sense. on it.